lte/gateway/c/oai/tasks/amf/amf_as.cpp

```diff
@@ -0,0 +1,1003 @@
+      M5GS_MOBILITY_MANAGEMENT_MESSAGE;
+
+  if (security && (security->ksi != KSI_NO_KEY_AVAILABLE)) {
+    /*
+     * A valid 5G CN security context exists
+     */
+    if (security->is_new) {
+      /*
+       * New 5G CN security context is taken into use
+       */
+      if (security->is_knas_int_present) {
+        if (security->is_knas_enc_present) {
+          /*
+           * NAS integrity and cyphering keys are available
+           */
+          msg->header.security_header_type =
+              SECURITY_HEADER_TYPE_INTEGRITY_PROTECTED_CYPHERED_NEW;
+        } else {
+          /*
+           * NAS integrity key only is available
+           */
+          msg->header.security_header_type =
+              SECURITY_HEADER_TYPE_INTEGRITY_PROTECTED_NEW;
+        }
+        OAILOG_FUNC_RETURN(LOG_NAS_AMF, &msg->security_protected.plain.amf);
+      }
+    } else if (security->is_knas_int_present) {
+      if (security->is_knas_enc_present) {
+        /*
+         * NAS integrity and cyphering keys are available
+         */
+        msg->header.security_header_type =
+            SECURITY_HEADER_TYPE_INTEGRITY_PROTECTED_CYPHERED;
+      } else {
+        /*
+         * NAS integrity key only is available
+         */
+        msg->header.security_header_type =
+            SECURITY_HEADER_TYPE_INTEGRITY_PROTECTED;
+      }
+      OAILOG_FUNC_RETURN(LOG_NAS_AMF, &msg->security_protected.plain.amf);
+    } else {
+      /*
+       * No valid 5G CN security context exists
+       */
+      msg->header.security_header_type = SECURITY_HEADER_TYPE_NOT_PROTECTED;
+      OAILOG_FUNC_RETURN(LOG_NAS_AMF, &msg->plain.amf);
+    }
+  } else {
+    /*
+     * No valid 5G CN security context exists
+     */
+    msg->header.security_header_type = SECURITY_HEADER_TYPE_NOT_PROTECTED;
+    OAILOG_FUNC_RETURN(LOG_NAS_AMF, &msg->plain.amf);
+  }
+
+  /*
+   * A valid 5G CN security context exists but NAS integrity key
+   * * * * is not available
+   */
+  OAILOG_FUNC_RETURN(LOG_NAS_AMF, NULL);
+}
+
+/***************************************************************************
+ **                                                                       **
+ ** Name:        amf_send_identity_request()                              **
+ **                                                                       **
+ ** Description: Send Identity Request to UE                              **
+ **                                                                       **
+ ** Inputs:      msg: Security msg                                        **
+ **              amf_msg :   amf msg                                      **
+ **                                                                       **
+ ** Return:   size                                                        **
+ **                                                                       **
+ **************************************************************************/
+static int amf_send_identity_request(
```

Comment:
This function is not sending the identity request message. Is the plan to add it in future PR?

Reply:
Fixed